Turn 'deployment considerations' into 'deployment path' (#3456)
theforeman/foreman-documentation: guides/common/modules/con_best-practices-for-deploying-smartproxies.adoc

Reviewer: please drop this empty file.

Author: Done.

[id="best-practices-for-deploying-{smart-proxy-context}_{context}"]

> = Best practices for deploying {SmartProxies}

Reviewer: ```suggestion
= Best practices for deploying {SmartProxyServers}
```

same for anchor and file name.

* Install a {SmartProxyServer} in each of your geographic locations.
By assigning a {SmartProxy} to each location, you decrease the load on {ProjectServer}, increase redundancy, and reduce bandwidth usage.
+
[NOTE]
====
The maximum number of {SmartProxyServers} that {ProjectServer} can support has no fixed limit.
It was tested that a {ProjectServer} can support 17 {SmartProxyServers} with 2 vCPUs.
====

> * Your {ProjectServer} contains an integrated {SmartProxyServer}.

Reviewer: ```suggestion
* Your {ProjectServer} contains an integrated {SmartProxy}.
```

I do not like the word "integrated" at all. However, I cannot provide a good/better alternative, so I'm OK.

> Do not use the integrated {SmartProxyServer} to communicate with hosts in remote locations because it can lead to suboptimal network utilization.

Reviewer: ```suggestion
Do not use the integrated {SmartProxy} to communicate with hosts in remote locations because it can lead to suboptimal network utilization.
```

I am unsure if this is correct/ if we should make this broad recommendation at this stage. Maybe drop this sentence and instead link to https://docs.theforeman.org/nightly/Tuning_Performance/index-katello.html#Smart_Proxy_Configuration_Tuning_performance-tuning?

Author: I think the whole 'best practices' section for deploying capsules might be pretty weak, actually. I merged it with the module on installing capsules. Adding a link to the tuning guide is a very good idea. Thanks!

> * Decide what services you want to enable on each {SmartProxy}.

Reviewer: ```suggestion
* Decide what services you want to enable on each {SmartProxyServer}.
```

* Decide whether you want to run DNS, DHCP, and TFTP on your {SmartProxyServers} or whether you want to configure your {SmartProxyServers} to use services provided by an external server.

.Additional resources

> * For more information about installing {SmartProxies}, including information about configuring a {SmartProxy} with external services, see link:{InstallingSmartProxyDocURL}[_{InstallingSmartProxyDocTitle}_].

Reviewer: ```suggestion
* For more information about installing {SmartProxyServer}, including information about configuring a {SmartProxyServer} with external services, see {InstallingSmartProxyDocURL}[_{InstallingSmartProxyDocTitle}_].
```
